src/textacy/extract/triples.py

```diff
@@ -217,86 +219,83 @@ def direct_quotations(doc: Doc) -> Iterable[DQTriple]:
         Loosely inspired by Krestel, Bergler, Witte. "Minding the Source: Automatic
         Tagging of Reported Speech in Newspaper Articles".
     """
-    # TODO: train a model to do this instead, maybe similar to entity recognition
     try:
         _reporting_verbs = constants.REPORTING_VERBS[doc.lang_]
     except KeyError:
         raise ValueError(
             f"direct quotation extraction is not implemented for lang='{doc.lang_}', "
             f"only {sorted(constants.REPORTING_VERBS.keys())}"
         )
-    qtok_idxs = [tok.i for tok in doc if tok.is_quote]
-    if len(qtok_idxs) % 2 != 0:
-        raise ValueError(
-            f"{len(qtok_idxs)} quotation marks found, indicating an unclosed quotation; "
-            "given the limitations of this method, it's safest to bail out "
-            "rather than guess which quotation is unclosed"
-        )
-    qtok_pair_idxs = list(itertoolz.partition(2, qtok_idxs))
-    for qtok_start_idx, qtok_end_idx in qtok_pair_idxs:
-        content = doc[qtok_start_idx : qtok_end_idx + 1]
+    # pairs up quotation-like characters based on acceptable start/end combos
+    # see constants for more info
+    qtoks = [tok for tok in doc if tok.is_quote or (re.match(r"\n", tok.text))]
+    qtok_idx_pairs = [(-1,-1)]
+    for n, q in enumerate(qtoks):
+        if (
+            not bool(q.whitespace_)
```

Comment:
guessing this is related to my earlier question about why we also allow tokens with newlines in them to be quotation tokens...

```suggestion
            not q.is_space
```

Reply:
`q.whitespace_` is whether or not the character has a trailing whitespace, not whether it is a space!

Reply:
ahh yes, i misremembered. all good here.